### PR TITLE
TSE: add fail detection and timeout

### PR DESCRIPTION
**Pull request by mhcp6**

### CODE REVIEW

File: stustapay/tse/tse_switchover.py [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+
+                print("done")
+
+            if self.show:
+                tses = await psql.fetch("select * from tse")
+                print("TSE master data:")
+                for tse in tses:
+                    for key, item in tse.items():
+                        print(f"{key}: {item}")
+                        print("-------------------------------------------------------")
```
Comment (TheJJ) [inline stustapay/tse/tse_switchover.py:260]: i'd leave out this line, rather move it out of the loop.

File: stustapay/tse/wrapper.py [hunk 2]
```diff
@@ -158,7 +256,8 @@ async def _tse_handler_loop(self):
                 LOGGER.info(f"signature result: {result!r}")
                 if result is None:
                     # the signature was cleanly aborted and should be reattempted.
-                    await self._return_request(next_request)
+                    # await self._return_request(next_request)
```
Comment (TheJJ) [inline stustapay/tse/wrapper.py:259]: leftover comment?

Comment (mhcp6) [reply stustapay/tse/wrapper.py]: cleaned

File: stustapay/tse/tse_switchover.py (hunk 1 above)
Comment (mhcp6) [reply stustapay/tse/tse_switchover.py:321]: done

File: stustapay/tse/diebold_nixdorf_usb/simulator.py [hunk 3]
```diff
@@ -283,6 +289,13 @@ def finishtrans(self, msg):
             return dnerror(5017)
         self.current_transactions[msg["ClientID"]].remove(msg["TransactionNumber"])
 
+        if self._broken:
+            if self.transnr >= 20:
+                self.transnr = 40
+                print("TIIIIIIIME TOOOOOOOOO SAY GOODBYE........")
+                while True:
```
[the remark this answers is not in the document]
Comment (mhcp6) [reply stustapay/tse/diebold_nixdorf_usb/simulator.py:296]: now sleep(1)

File: stustapay/tse/tse_switchover.py [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+                    await psql.execute("update tse set tse_status='failed' where tse_name=$1", self.tse_to_disable)
+
+                # print assigned tills
+                print("These tills are currently assigned to this TSE:\n\n")
+                tills = await psql.fetch("select id, name from till where tse_id=$1 order by id", tse["tse_id"])
+                for till in tills:
+                    print(f"{till['id']:5}:  {till['name']:50}")
+
+                print("ALL OF THESE TILLS NEED TO BE LOGGED OUT!")
+                input("confirm with enter when ready...")
```
[the remark this answers is not in the document]
Comment (mhcp6) [reply stustapay/tse/tse_switchover.py]: extra checks for logout and forced logout option introduced

**Review by TheJJ — APPROVED** [review]

**Pull request merged**